### PR TITLE
Fix ReadableParcelAndroid::ReadString interface

### PR DESCRIPTION
**Pull request by sifmelcara**
Also uses implementation from android/binder_parcel_utils.h to read
ByteArray and string from Parcel

Test example app on device, works correctly

### CODE REVIEW

File: src/core/ext/transport/binder/wire_format/binder_android.cc
Comment (TaWeiTu) [inline src/core/ext/transport/binder/wire_format/binder_android.cc:287]: Can we use `vec.data()` here?

**Review by TaWeiTu — APPROVED** [review]
LGTM modulo comments, but this will probably be rewritten again if we are going to optimize the receiving part and use `grpc_slice` all the way through (directly reading into the `grpc_slice` buffer).

File: src/core/ext/transport/binder/wire_format/binder_android.cc
Comment (sifmelcara) [reply src/core/ext/transport/binder/wire_format/binder_android.cc:287]: Done. Also added a guard to skip memcpy when size is 0

**Comment by sifmelcara** [comment]
Rebased. Since all tests passed before, will enable auto merge

**Pull request merged**
